internal/api/pkce.go

```diff
@@ -78,3 +79,16 @@ func getFlowFromChallenge(codeChallenge string) models.FlowType {
 		return models.ImplicitFlow
 	}
 }
+
+func generateFlowStateIfPKCE(flowType models.FlowType, authenticationMethod models.AuthenticationMethod, codeChallengeMethodParam string, codeChallenge string, userID *uuid.UUID) (*models.FlowState, error) {
```

Comment:
probably can leave the `isPKCE` part of it external to the function.  Will do that tomorrow